userdocs/diagnostics/diagnostics.md

```diff
@@ -4,4 +4,84 @@
    @TechnologyRoot
 }
 
-Documentation on diagnostics emitted by the compiler.
\ No newline at end of file
+Documentation on diagnostics emitted by the compiler and settings that control them.
+
+
+## Diagnostics
+
+Swift diagnostics are classified into errors and warnings. Warnings can only be silenced within
+source in an intentional manner, eg. adding `_ =` for an unused function result.
+
+Some diagnostics have extra documentation, these include a `[#Name]` inline and reference to this
+documentation at the end of the compiler output on the command line, or is presented specially
+within your IDE of choice. See below for the full list of these notes.
+
+
+## Diagnostic groups
+
+Diagnostic groups collect some number of diagnostics together under a common group name. This allows
+for extra documentation (see the list below) to help explain relevant language concepts, as well as
+the ability to control the behavior of warnings in a more precise manner:
+- `-Werror <group>` - upgrades warnings in the specified group to errors
+- `-Wwarning <group>` - indicates that warnings in the specified group should remain warnings, even
+  if they were previously upgraded to errors
+
+As a concrete example, to upgrade deprecated declaration warnings to errors:
+```sh
+-Werror DeprecatedDeclaration
+```
+
+Or upgrade all warnings except deprecated declaration to errors:
+```sh
+-warnings-as-errors -Wwarning DeprecatedDeclaration
+```
+
+
+## Upcoming language features
+
+Upcoming language features allow for enabling features that will become the default in an upcoming
+language mode in your current language mode. This allows the incremental adoption of language
+features without having to fully migrate to a new language mode. They can be enabled on the command
+line with `-enable-upcoming-feature <feature>`.
+
+
+## Topics
+
+### Diagnostics list
+
+- <doc:actor-isolated-call>
+- <doc:conformance-isolation>
+- <doc:dynamic-callable-requirements>
+- <doc:error-in-future-swift-version>
+- <doc:existential-member-access-limitations>
+- <doc:isolated-conformances>
+- <doc:member-import-visibility>
+- <doc:multiple-inheritance>
+- <doc:mutable-global-variable>
+- <doc:nominal-types>
+- <doc:opaque-type-inference>
+- <doc:property-wrapper-requirements>
+- <doc:protocol-type-non-conformance>
+- <doc:result-builder-methods>
+- <doc:sendable-closure-captures>
+- <doc:sending-closure-risks-data-race>
+- <doc:sending-risks-data-race>
+- <doc:string-interpolation-conformance>
+- <doc:temporary-pointers>
+- <doc:trailing-closure-matching>
```

Comment:
Should we either generate this via a script or add a regression test to verify that all of the markdown files end up somewhere in this file?